scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+Usage() {
+    cat <<EOF
+Usage:
+    $0 <package_git_directory> <package_lookaside_directory>
+
+    Removes all tarballs in <package_lookaside_directory> except
+    for tarballs that are referenced by the latest commit of each
+    branch in <package_git_directory>.
+EOF
+}
+
+if [[ $# != 2 ]]; then
+    Usage
+    exit 1
+fi
+
+pkg_git_dir="$1"
+pkg_lookaside_dir="$2"
+
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_lookaside_dir" ]; then
+    echo "$pkg_lookaside_dir is not a valid directory."
+    exit 1
+fi
+
+pushd "$pkg_git_dir" > /dev/null
```

Comment:
shellcheck: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
(use pushd || exit 1)